### PR TITLE
some jsdoc fixes

### LINKED ISSUE #2869: Write custom lint rules
- [x] Make sure `warning` and `invariant` have a string literal for first argument
- [ ] Make sure JSDoc args match up

### PR DESCRIPTION
**Pull request by hzoo**
Similar to #2772.

Will need some help to make sure the changes make sense!

I'm using these jscs rules right now to find issues

```js
"jsDoc": {
   "checkRedundantReturns": true, //check @return with if anything is actually returned
   "checkParamNames": true //make sure parameter names match the @param name
}
```

### CODE REVIEW

File: src/utils/Transaction.js [hunk 1]
```diff
@@ -107,12 +107,18 @@ var Mixin = {
...
    * methods that result in large amounts of computation/mutations that would
-   * need to be safety checked.
+   * need to be safety checked. The optional arguments helps prevent the need
+   * to bind in many cases.
    *
    * @param {function} method Member of scope to call.
    * @param {Object} scope Scope to invoke from.
-   * @param {Object?=} args... Arguments to pass to the method (optional).
-   *                           Helps prevent need to bind in many cases.
+   * @param {Object?=} a Arguments to pass to the method.
```
Comment (hzoo) [inline src/utils/Transaction.js]: not sure if this is good or not

File: src/browser/eventPlugins/MobileSafariClickEventPlugin.js [hunk 2]
```diff
@@ -35,7 +35,6 @@ var MobileSafariClickEventPlugin = {
    * @param {DOMEventTarget} topLevelTarget The listening component root node.
    * @param {string} topLevelTargetID ID of `topLevelTarget`.
    * @param {object} nativeEvent Native browser event.
-   * @return {*} An accumulation of synthetic events.
```
Comment (hzoo) [inline src/browser/eventPlugins/MobileSafariClickEventPlugin.js]: It doesn't look like anything is being returned

File: src/classic/element/ReactElement.js [hunk 3]
```diff
@@ -85,8 +85,10 @@ function defineMutationMembrane(prototype) {
  * work with a dynamic instanceof check. Nothing should live on this prototype.
  *
  * @param {*} type
- * @param {string|object} ref
  * @param {*} key
+ * @param {string|object} ref
```
Comment (hzoo) [inline src/classic/element/ReactElement.js]: missing params

File: src/test/createHierarchyRenderer.js [hunk 4]
```diff
@@ -47,7 +47,7 @@ var React = require('React');
  * each invocation of the hierarchy renderer. If new classes are needed, you
  * should re-execute `createHierarchyRenderer` with the same arguments.
  *
- * @param {array<function>} ...renderMethods
+ * @param {array<function>} renderMethods
```
Comment (hzoo) [inline src/test/createHierarchyRenderer.js]: I guess the param is still called `renderMethods` even if it has a spread operator

File: src/utils/OrderedMap.js [hunk 5]
```diff
@@ -97,7 +97,7 @@ function assertValidPublicKey(key) {
  * @param {number} length Length of range.
  * @param {number} actualLen Actual length of range that should not be
  * exceeded.
- * @return {void} description
+ * @throws Error if range arguments are out of bounds.
```
Comment (hzoo) [inline src/utils/OrderedMap.js]: no return so throws?

**Comment by hzoo** [comment]
rebased to latest

**Comment by hzoo** [comment]
Fixed some more issues and rebased: cc @zpao

**Comment by zpao** [comment]
Cool. I think this should be fine. I want to do this one specially since it's internal infra that breaks, not public. So a bit more delay since it requires more work.

**Comment by hzoo** [comment]
Ok will do! There's probably a few more things I missed but there's a lot already to go through for now.

**Comment by hzoo** [comment]
This still good? Will need to redo

**Comment by hzoo** [comment]
any updates on this? @zpao

**Comment by zpao** [comment]
Sorry. Yea, we can have another go at this.

**Comment by hzoo** [comment]
Ok I rebased to master.

File: src/event/eventPlugins/ResponderEventPlugin.js [hunk 6]
```diff
@@ -570,7 +570,7 @@ var ResponderEventPlugin = {
 
   injection: {
     /**
-     * @param {{onChange: (ReactID, ReactID) => void} GlobalResponderHandler
+     * @param {{onChange: (ReactID, ReactID) => void}} GlobalResponderHandler
```
Comment (hzoo) [inline src/event/eventPlugins/ResponderEventPlugin.js]: Not sure if this is right.

File: src/utils/OrderedMap.js [hunk 7]
```diff
@@ -438,7 +438,7 @@ var OrderedMapMethods = {
    * because `undefined` is not ever a valid `OrderedMap` key.
    *
    * @private
-   * @param {number} pos Position for which we're querying the name.
+   *  @param {number} pos Position for which we're querying the name.
```
Comment (hzoo) [inline src/utils/OrderedMap.js]: Not sure about this - could remove this line?

File: src/event/eventPlugins/ResponderEventPlugin.js (hunk 6 above)
Comment (zpao) [inline src/event/eventPlugins/ResponderEventPlugin.js]: I don't think an object spec like this is even valid jsdoc so it probably doesn't matter.

File: src/utils/OrderedMap.js (hunk 7 above)
Comment (zpao) [inline src/utils/OrderedMap.js]: Yea, let's remove it.

**Comment by zpao** [comment]
Awesome, if we can get those updates in & squash the commits, we can get this merged.

File: src/event/eventPlugins/ResponderEventPlugin.js (hunk 6 above)
Comment (hzoo) [inline src/event/eventPlugins/ResponderEventPlugin.js]: Yeah like the arrow function xp. Should I leave it?

File: src/utils/OrderedMap.js (hunk 7 above)
Comment (hzoo) [inline src/utils/OrderedMap.js]: Ok and what about the comments above? "Finds the key at a given position..."

**Comment by hzoo** [comment]
Ok updated

**Comment by hzoo** [comment]
ping @zpao

**Comment by zpao** [comment]
Merged it manually to fix a couple new conflicts. Sorry for the delay!

**Comment by hzoo** [comment]
:smile: cool!

**Pull request merged**